Logtail.NLog/LogtailTarget.cs

```diff
@@ -45,24 +43,54 @@ public sealed class LogtailTarget : TargetWithContext, IUsesStackTrace
         /// We capture the file and line of every log message by default. You can turn this
         /// option off if it has negative impact on the performance of your application.
         /// </summary>
-        public bool CaptureSourceLocation { get; set; } = true;
+        public bool CaptureSourceLocation
+        {
+            get => StackTraceUsage == StackTraceUsage.Max;
+            set => StackTraceUsage = value ? StackTraceUsage.Max : StackTraceUsage.WithoutSource;
+        }
 
         /// <summary>
         /// Include GlobalDiagnosticContext in logs.
         /// </summary>
         public bool IncludeGlobalDiagnosticContext { get; set; } = true;
 
-        public StackTraceUsage StackTraceUsage { get; private set; } = StackTraceUsage.Max;
+        public StackTraceUsage StackTraceUsage
+        {
+            get => _stackTraceUsage ?? StackTraceUsage.Max;
+            set
+            {
+                if (value == StackTraceUsage.None)
+                {
+                    IncludeCallSite = false;
+                    IncludeCallSiteStackTrace = false;
+                }
+                else
+                {
+                    IncludeCallSite = true;
+                    IncludeCallSiteStackTrace = value == StackTraceUsage.Max;
+                }
+                _stackTraceUsage = value;
+            }
+        }
+        private StackTraceUsage? _stackTraceUsage = null;
 
         private Drain logtail = null;
 
         protected override void InitializeTarget()
         {
+            if (!_stackTraceUsage.HasValue)
+            {
+                StackTraceUsage = StackTraceUsage.Max;
+            }
+
             logtail?.Stop().Wait();
 
+            var sourceToken = RenderLogEvent(SourceToken, LogEventInfo.CreateNullEvent());
+            var endpoint = RenderLogEvent(Endpoint, LogEventInfo.CreateNullEvent());
```

Comment:
Could you explain why these are necessary? Is it just so that they are turned from `Layout` into string? In that case, do we need to define these as `Layout` in the first place? Just asking 🙂

Reply:
The NLog `Layout` allows you to retrieve values from configuration-files like `app.config` or `appsettings.json`. So the same NLog.config file can be used on all environments, and any enviroment-specific-details can be applied from "outside configuration".